### PR TITLE
[7.x] Update compiler plugin to allow annotations

### PR DESCRIPTION
**Pull request by aashikam**
## Purpose
fixes https://github.com/ballerina-platform/ballerina-standard-library/issues/4734

## Checklist
- [x] Linked to an issue
- [x] Updated the changelog
- [x] Added tests
- [ ] Updated the spec
- [ ] Checked native-image compatibility

### CODE REVIEW

File: compiler-plugin-tests/src/test/java/io/ballerina/stdlib/rabbitmq/plugin/RabbitmqCompilerPluginTest.java [hunk 1]
```diff
@@ -127,6 +127,14 @@ public void testValidService9() {
         Assert.assertEquals(diagnosticResult.errorCount(), 0);
     }
 
+    @Test(enabled = true, description = "Validate `kafka:Service` with `display` annotation")
```
Comment (dilanSachi) [inline compiler-plugin-tests/src/test/java/io/ballerina/stdlib/rabbitmq/plugin/RabbitmqCompilerPluginTest.java:130]:
```suggestion
    @Test(enabled = true, description = "Validate `rabbitmq:Service` with `display` annotation")
```

File: compiler-plugin/src/main/java/io/ballerina/stdlib/rabbitmq/plugin/RabbitmqServiceValidator.java [hunk 2]
```diff
@@ -99,45 +97,24 @@ private void validateAttachPoint(SyntaxNodeAnalysisContext context) {
...
-                    } else {
-                        validateAnnotation(symbolList.get(0), serviceDeclarationNode.location(), context);
-                    }
-                }
+            Optional<ServiceAttachPoint> serviceNameAttachPoint = serviceDeclarationSymbol.attachPoint();
+            List<AnnotationSymbol> annotations = serviceDeclarationSymbol.annotations();
+
+            // RabbitMQ gets the subject name from either the service name or the
+            // service config, so either one of them should be present.
+            if (serviceNameAttachPoint.isEmpty() && annotations.isEmpty()) {
```
Comment (ThisaruGuruge) [inline compiler-plugin/src/main/java/io/ballerina/stdlib/rabbitmq/plugin/RabbitmqServiceValidator.java:105]: What if the `serviceNameAttachPoint` is empty and there is a `@display` annotation, without the service config? It should be an error, but this does not catch it, right?

File: compiler-plugin-tests/src/test/resources/ballerina_sources/valid_service_10/service.bal [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerinax/rabbitmq;
+
+listener rabbitmq:Listener channelListener =
+        new(rabbitmq:DEFAULT_HOST, rabbitmq:DEFAULT_PORT);
+
+@display {
+    label: "natsService"
```
Comment (dilanSachi) [inline compiler-plugin-tests/src/test/resources/ballerina_sources/valid_service_10/service.bal:23]:
```suggestion
    label: "rabbitmqService"
```

File: compiler-plugin/src/main/java/io/ballerina/stdlib/rabbitmq/plugin/RabbitmqServiceValidator.java (hunk 2 above)
Comment (aashikam) [reply compiler-plugin/src/main/java/io/ballerina/stdlib/rabbitmq/plugin/RabbitmqServiceValidator.java:105]: hmm, yeah have to add a validation to check if the annotation is service config

File: compiler-plugin-tests/src/test/java/io/ballerina/stdlib/rabbitmq/plugin/RabbitmqCompilerPluginTest.java (hunk 1 above)
Comment (ThisaruGuruge) [inline compiler-plugin-tests/src/test/java/io/ballerina/stdlib/rabbitmq/plugin/RabbitmqCompilerPluginTest.java:130]: We don't need `enabled = true` here. It is enabled by default.

File: compiler-plugin/src/main/java/io/ballerina/stdlib/rabbitmq/plugin/RabbitmqServiceValidator.java (hunk 2 above)
Comment (aashikam) [reply compiler-plugin/src/main/java/io/ballerina/stdlib/rabbitmq/plugin/RabbitmqServiceValidator.java]: Fixed in https://github.com/ballerina-platform/module-ballerinax-rabbitmq/pull/874/commits/99cee71894b3171feb63b5e5b96c6e75342d695c

File: compiler-plugin/src/main/java/io/ballerina/stdlib/rabbitmq/plugin/RabbitmqServiceValidator.java [hunk 4]
```diff
@@ -99,45 +98,48 @@ private void validateAttachPoint(SyntaxNodeAnalysisContext context) {
...
+            if (moduleSymbolOptional.isPresent()) {
+                ModuleSymbol moduleSymbol = moduleSymbolOptional.get();
+                if (moduleSymbol.id().orgName().equals(PluginConstants.PACKAGE_ORG) ||
+                        moduleSymbol.id().moduleName().equals(PluginConstants.PACKAGE_PREFIX)) {
+                    // not checking name as rabbitmq has only two annotations and only one is allowed on services.
+                    flag = true;
+                }
             }
         }
+        return flag;
```
Comment (ThisaruGuruge) [inline compiler-plugin/src/main/java/io/ballerina/stdlib/rabbitmq/plugin/RabbitmqServiceValidator.java:139]: * No need for a separate `flag` variable. We can directly return `true` if we find the correct annotation. No need to iterate through the whole list, if we found the correct one.
* When using the `equals` check, always better to use the `equals()` method on the constant, will prevent NPEs. 
* (Subjective take) I don't think using `Optional<>` variables is a good thing. Returning `Optional` is one thing, but assigning variables to `Optional` doesn't look nice.
* The inner-most `if` condition should be `&&`, not `||`, right? We should check whether both the org and module name are matching. 

```suggestion
        for (AnnotationSymbol annotationSymbol : annotationSymbols) {
            if (annotationSymbol.getModule().isPresent()) {
                ModuleSymbol moduleSymbol = annotationSymbol.getModule().get();
                if (PluginConstants.PACKAGE_ORG.equals(moduleSymbol.id().orgName()) &&
                        PluginConstants.PACKAGE_PREFIX.equals(moduleSymbol.id().moduleName())) {
                    // not checking name as rabbitmq has only two annotations and only one is allowed on services.
                    return true;
                }
            }
        }
        return false;
```

**Review by ThisaruGuruge — APPROVED** [review]

**Pull request merged**